homeassistant/components/recorder/statistics.py

```diff
@@ -2221,9 +2283,75 @@ def _import_statistics_with_session(
         else:
             _insert_statistics(session, table, metadata_id, stat)
 
+    if table != StatisticsShortTerm:
+        return True
+
+    # We just inserted new short term statistics, so we need to update the
+    # latest_statistics_short_term_ids table that tracks what the newest id is
+    # for the metadata_id.
+    session.execute(
+        _delete_latest_statistics_short_term_ids_by_metadata_ids_stmt({metadata_id}),
+    )
+    if latest := cast(
+        Sequence[Row],
+        execute_stmt_lambda_element(
+            session, _find_latest_short_term_statistic_stmt(metadata_id), orm_rows=False
+        ),
+    ):
+        session.add(
+            LatestStatisticsShortTermIDs(metadata_id=metadata_id, id=latest[0].id)
+        )
+
     return True
 
 
+def _find_latest_short_term_statistic_stmt(
+    metadata_id: int,
+) -> StatementLambdaElement:
+    """Find the latest short term statistics for a list of metadata_ids."""
+    return lambda_stmt(
+        lambda: select(
+            StatisticsShortTerm.id,
+        )
+        .where(StatisticsShortTerm.metadata_id == metadata_id)
+        .order_by(StatisticsShortTerm.start_ts.desc())
+        .limit(1)
+    )
+
+
+def _delete_latest_statistics_short_term_ids_by_metadata_ids_stmt(
+    metadata_ids: set[int],
+) -> StatementLambdaElement:
+    """Delete LatestStatisticsShortTermIDs rows.
```

Comment:
Can we be consistent in the docstrings, either we use the language as in this function everywhere "Do something" or we use the language "Create the statement which does something".

I much prefer the latter option so it's clear from the docstring the function itself does not have side effects, but just prepares a query.

Reply:
code has been deleted in refactoring